### DISCUSSION TITLE
How do I pass arguments to main.py that is run by uvicorn

### DISCUSSION THREAD

**Discussion by ravi567** [fastapi/fastapi#9322] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# main.py

from fastapi import FastAPI
import getopt, configparser

try:   
     opts , args = getopt.getopt ( argv, " f:l: " )   
        
except:   
     print ( " Error " )
for opt , arg in opts:   
     if opt in ("-e", "--env"):
          my_env = arg

config = configparser.ConfigParser()
config.read('system.conf')
db_user = config[my_env]['db_user']

app = FastAPI()

@app.get("/")
async def root():
    return {"message": "Hello World"}
```

### Description

I am using FastAPI to expose a rest endpoint that takes data and forwards it to a database table. The script needs to connect to different databases depending on the environment (development, testing, production etc).  All other scripts for the application I am building use a -e <env name> as an argument that is parsed by getopt. I then use configparser to read the required values from a conf file. Is there a way to pass arguments to the python script used by uvicorn?

### Operating System

Linux

### FastAPI Version

0.95.0

### Python Version

3.10

**Comment by rlimberger** [comment]
Did you ever find an answer to this? Im wondering the same.

**Comment by CroMarmot** [comment]
i also wonder how, looking around issues and docs, they all use `env variable` or `env file`,

here is a **wierd** but workable solution. **use flask** instead of uvicorn 😥


```py
from a2wsgi import ASGIMiddleware
from werkzeug.middleware.dispatcher import DispatcherMiddleware
from flask import Flask
from fastapi import FastAPI

def func():
    flask_app = Flask(__name__)
    fastapi_app = FastAPI(root_path="/api/fastapi")

    flask_app.wsgi_app = DispatcherMiddleware(
        flask_app.wsgi_app,
        {
            "/api/flask": app,
            "/api/fastapi": ASGIMiddleware(fastapi_app),
        },
    )
    flask_app.run(host="0.0.0.0", port=5000, debug=True)
```

in this way all code are running in same memory 😥

**Comment by YuriiMotov** [comment, marked as answer]
You can use ENV variables:

```python
import os

my_config = os.getenv("MY_CONFIG")
```

Provide ENV variables when you run uvicorn:
```
MY_CONFIG="value" uvicorn main:app
```
